dotcom-rendering/src/components/TableBlockComponent.tsx

```diff
@@ -1,54 +1,113 @@
 
+const tableStyles = css`
+	width: 100%;
+	background: ${palette('--table-block-background')};
+	border-top: 0.0625rem solid ${palette('--table-block-border-top')};
+	color: ${palette('--table-block-text')};
+	border-collapse: inherit;
+`;
+
+const headStyles = css`
+	tr {
+		font-weight: 800;
+		text-align: left;
+	}
+`;
+
+const rowStyles = css`
+	${textSans12};
+	:nth-child(odd) > td {
+		background-color: ${palette('--table-block-stripe')};
+	}
+`;
+
+const cellPadding = css`
+	padding: 0.5rem;
+`;
+
 const tableEmbed = css`
-	.table--football {
-		width: 100%;
-		background: ${palette.neutral[97]};
-		border-top: 0.0625rem solid ${palette.focus[400]};
-		color: ${palette.neutral[7]};
-		border-collapse: inherit;
-		tr:nth-child(odd) > td {
-			background-color: ${palette.neutral[93]};
+	tr > th:first-child,
+	td:first-child {
+		color: ${palette('--table-block-text-first-column')};
+	}
+`;
+
+type Props = {
+	element: TableBlockElement;
+};
+
+const buildElementTree = (node: Node) => {
+	const children = Array.from(node.childNodes).map(buildElementTree);
+	switch (node.nodeName) {
+		case 'TABLE': {
+			return (
+				<table className="table--football" css={tableStyles}>
+					{children}
+				</table>
+			);
 		}
-		th {
-			padding: 0.5rem;
+		case 'THEAD': {
+			return <thead css={headStyles}>{children}</thead>;
 		}
-		td {
-			padding: 0.5rem;
+		case 'TBODY': {
+			return <tbody>{children}</tbody>;
 		}
-		tr {
-			${textSans12};
+		case 'ABBR': {
+			return (
+				<abbr title={(node as HTMLElement).getAttribute('title') ?? ''}>
+					{children}
+				</abbr>
+			);
 		}
-		thead {
-			tr {
-				font-weight: 800;
-				text-align: left;
-			}
+		case 'TR': {
+			return <tr css={rowStyles}>{children}</tr>;
 		}
-		tr > th:first-child,
-		td:first-child {
-			color: ${text.supporting};
+		case 'TH': {
+			return <th css={cellPadding}>{children}</th>;
 		}
-		.table-column--main {
-			width: 100%;
+		case 'TD': {
+			const isMainColumn = (node as HTMLElement).className.includes(
+				'table-column--main',
+			);
+			return (
+				<td
+					style={isMainColumn ? { width: '100%' } : undefined}
+					css={cellPadding}
+				>
+					{children}
+				</td>
+			);
 		}
+		case '#text': {
+			return node.textContent;
+		}
+		default:
+			logger.warn('TableBlockComponent: Unknown element received', {
+				isDev: process.env.NODE_ENV !== 'production',
+				element: {
+					name: node.nodeName,
+					html: isElement(node) ? node.outerHTML : undefined,
+				},
+			});
+			return null;
 	}
-	margin-bottom: 16px;
-`;
-
-type Props = {
-	element: TableBlockElement;
 };
 
 export const TableBlockComponent = ({ element }: Props) => {
+	const fragment = parseHtml(element.html);
 	return (
 		<div
 			css={tableEmbed}
+			style={{ marginBottom: '16px' }}
 			data-testid="football-table-embed"
-			dangerouslySetInnerHTML={{ __html: unescapeData(element.html) }}
-		/>
+		>
+			{Array.from(fragment.childNodes).map(buildElementTree)}
+		</div>
```

Comment:
Nice avoidance of dangerously setting inner html 👏 Much safer

In future we should look to send only the data from frontend, rather than all the HTML. This is a good refactor ahead of that.